### DISCUSSION TITLE
Rather a lot of SPAM

### DISCUSSION THREAD

**Discussion by graemev** [forwardemail/forwardemail.net#295] [General]
I run my own spam filters on the mail I POP from forwardemail. I keep the spam, mostly so it can be used as training data. Yesterday I noticed my mailserver was low on disk space . I discovered 2.4Gb of SPAM.

The point of interest is, to log goes back to Mar 2019, there are 74,667 spam messages, however 70,905 have happened in 2024, since I switched to forwardemail.  So in 5 years of mail ...96% of the SPAM has arrived in the past few months

**Comment by graemev** [comment]
BTW, I found out why there was so much SPAM. There are a few addresses that have been compromised (sites hacked etc) that are now 100% spam . So I directed these into a SPAM mailbox on forwardemail.
Then , frequently (every-time forwardemail reboot an imap server and lose my connection) I MOVE these into my local (home) spambox ...however it appears forwardemail treats these "**move**" as "**copy**" so every hour or so I get another copy of the last months SPAM.

**Reply by titanism** [reply]
If you still have this MOVE / COPY issue let us know?  Try it out for a few days and if it still happens please do let us know.

**Reply by graemev** [reply]
OK, I've re-enabled the "MOVE" of spam . I'll keep eye on it.

**Comment by graemev** [comment]
Looks like it's still doing a copy , when I request a move:

3 matches for "SPAM These are all the UIDs being moved " in buffer: syslog                                                                                                                                     
  50266:2024-11-01T18:19:45.161806+00:00 zbox imapfilter[1801554]: SPAM These are all the UIDs being moved       =>[124] uids={list:111,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,18$
  51209:2024-11-01T19:18:38.381251+00:00 zbox imapfilter[1807412]: SPAM These are all the UIDs being moved       =>[124] uids={list:111,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,18$
  51573:2024-11-01T19:56:21.036378+00:00 zbox imapfilter[1811077]: SPAM These are all the UIDs being moved       =>[124] uids={list:111,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,18$

So over 3 hours , it MOVED the same 3 sets of mails from your server to mine.


[ I will remove the MOVE , now as its filling my spam box quite quickly ]

-- NO LONGER (07Oct2024) move_spam(account,home); ... put this back 01Nov24 ..they think it's fixed                                                                                                            
move_spam(account,home);


...

-- Take old spam off the external account                                                                                                                                                                      
function move_spam(from,to)

   frombox   = from.spam;
   tobox     = to.spam;

   msgs = frombox:is_older(parms.maxspamage);

   if options.info == true then
      local s=from._account.username .. "@" ..  from._account.server.. "/" .. "spam"
      local t=to._account.username .. "@" ..  to._account.server.. "/" .. "spam"

      if parms.logging > 0 then
         print ("SPAM Will move msgs from " .. s .. " to " .. t);

         print ("SPAM These are all the UIDs being moved       =>" .. str_uids(msgs));
      end
   end


   msgs:move_messages(tobox);

--   frombox.close();                                                                                                                                                                                          
--   tobox.close();                                                                                                                                                                                            
end

(I believe I tested this between my own mail servers)

**Comment by graemev** [comment]
ABTW, it still appears to lose the connection about once per hour:



  37964:2024-10-31T14:52:10.858360+00:00 zbox imapfilter[1638548]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  38165:2024-10-31T15:39:18.465137+00:00 zbox imapfilter[1641729]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  38763:2024-10-31T16:38:24.713631+00:00 zbox imapfilter[1646219]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  39596:2024-10-31T18:00:16.019918+00:00 zbox imapfilter[1651869]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  40031:2024-10-31T19:33:36.185566+00:00 zbox imapfilter[1659776]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  40330:2024-10-31T20:07:10.457799+00:00 zbox imapfilter[1668671]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  40475:2024-10-31T20:20:12.983658+00:00 zbox imapfilter[1671848]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  40684:2024-10-31T20:28:52.447877+00:00 zbox imapfilter[1673086]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  41046:2024-10-31T21:06:47.578522+00:00 zbox imapfilter[1673902]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  41278:2024-10-31T21:42:37.620009+00:00 zbox imapfilter[1677605]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  41865:2024-10-31T22:26:31.635588+00:00 zbox imapfilter[1681384]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  42265:2024-10-31T23:00:27.642343+00:00 zbox imapfilter[1685811]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  42370:2024-10-31T23:30:09.937004+00:00 zbox imapfilter[1689084]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  42609:2024-11-01T00:00:23.401753+00:00 zbox imapfilter[1691932]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  43186:2024-11-01T06:36:39.156488+00:00 zbox imapfilter[1729719]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  43391:2024-11-01T07:08:51.469499+00:00 zbox imapfilter[1732853]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  43789:2024-11-01T07:41:29.883809+00:00 zbox imapfilter[1736042]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  43948:2024-11-01T08:13:56.228609+00:00 zbox imapfilter[1739291]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  44281:2024-11-01T08:46:30.656199+00:00 zbox imapfilter[1742395]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  44452:2024-11-01T09:18:58.057894+00:00 zbox imapfilter[1745543]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  44962:2024-11-01T09:51:47.006369+00:00 zbox imapfilter[1748691]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  45177:2024-11-01T10:24:12.482472+00:00 zbox imapfilter[1751875]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  45548:2024-11-01T10:56:46.050885+00:00 zbox imapfilter[1754990]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  45787:2024-11-01T11:29:37.207506+00:00 zbox imapfilter[1758139]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  46109:2024-11-01T12:00:28.972859+00:00 zbox imapfilter[1761406]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  46262:2024-11-01T12:13:10.146757+00:00 zbox imapfilter[1764498]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  46707:2024-11-01T12:45:37.357711+00:00 zbox imapfilter[1765721]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  47086:2024-11-01T13:18:28.880203+00:00 zbox imapfilter[1768938]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  47246:2024-11-01T13:51:23.935660+00:00 zbox imapfilter[1772158]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  47421:2024-11-01T14:23:54.751609+00:00 zbox imapfilter[1775312]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  47637:2024-11-01T14:57:03.634236+00:00 zbox imapfilter[1778432]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  48035:2024-11-01T15:30:34.185731+00:00 zbox imapfilter[1781645]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  48298:2024-11-01T16:03:29.088008+00:00 zbox imapfilter[1784906]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  48661:2024-11-01T16:36:47.692310+00:00 zbox imapfilter[1788083]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  49064:2024-11-01T17:10:38.940092+00:00 zbox imapfilter[1791355]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  49493:2024-11-01T17:44:20.420644+00:00 zbox imapfilter[1794755]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  50171:2024-11-01T18:18:31.300023+00:00 zbox imapfilter[1798097]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  51151:2024-11-01T19:17:14.564607+00:00 zbox imapfilter[1801554]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  51471:2024-11-01T19:55:01.496072+00:00 zbox imapfilter[1807412]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  51820:2024-11-01T20:43:09.678715+00:00 zbox imapfilter[1811077]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  52050:2024-11-01T21:18:56.225710+00:00 zbox imapfilter[1815738]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed
  52198:2024-11-01T21:52:20.615494+00:00 zbox imapfilter[1819171]: imapfilter: idle request to graeme@vetterlein.com@imap.forwardemail.net failed

**Comment by titanism** [comment]
@graemev Can you please share your full scripts in code block syntax format?  Also, what version and package are you using for `imapfilter`?  Please send links to all software being used if possible, as well as gists or snippets to the full scripts you are using to MOVE.   Additionally, if you have error logs from when `idle request to xxx failed` and any response back from once `MOVE` is complete?  I'm not sure if the software you're using will purge the messages once MOVE is complete or not, as typically MOVE is specific to a single mailbox.
